src/dotnet/commands/dotnet-build/CompilerIOManager.cs

```diff
@@ -92,27 +93,28 @@ private CompilerIO ComputeIO(ProjectGraphNode graphNode)
             return new CompilerIO(inputs, outputs);
         }
 
-        private static void AddLockFile(ProjectContext project, List<string> inputs)
+        private static void AddLockFile(Project project, List<string> inputs)
         {
-            if (project.LockFile == null)
-            {
-                var errorMessage = $"Project {project.ProjectName()} does not have a lock file.";
-                Reporter.Error.WriteLine(errorMessage);
-                throw new InvalidOperationException(errorMessage);
-            }
-
-            inputs.Add(project.LockFile.LockFilePath);
-
-            if (project.LockFile.ExportFile != null)
-            {
-                inputs.Add(project.LockFile.ExportFile.ExportFilePath);
-            }
+            //TODO: better way to fix this:
+            //if (project.LockFile == null)
+            //{
+            //    var errorMessage = $"Project {project.ProjectName()} does not have a lock file.";
+            //    Reporter.Error.WriteLine(errorMessage);
+            //    throw new InvalidOperationException(errorMessage);
+            //}
+
+            //inputs.Add(project.LockFile.LockFilePath);
+
+            //if (project.LockFile.ExportFile != null)
```

Comment:
the name of the file is ExportFile.ExportFileName. By convention, it stays near the lock file. When it exists.

---
In reply to: [61284193](https://github.com/dotnet/cli/pull/2711#discussion_r61284193) [](ancestors = 61284193)